src/components/NoHouseholdIdModal.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+import type { UpdatePolicyRequestBody } from 'data/policies'
+import { Dialog, setNotice, TextField } from '@silintl/ui-components'
+import { createEventDispatcher } from 'svelte'
+
+export let open = false
+
+const policyData: UpdatePolicyRequestBody = {}
+
+let title = 'Missing information'
+let householdId = ''
+let hasClosed: boolean = false
+
+const buttons: Dialog.AlertButton[] = [
+  { label: 'Go Back', action: 'cancel', class: 'mdc-dialog__button' },
+  { label: 'Submit', action: 'submit', class: 'mdc-button--raised' },
+]
+
+const dispatch = createEventDispatcher<{ closed: { choice: string; data?: UpdatePolicyRequestBody } }>()
+
+const handleDialog = (event: CustomEvent) => {
+  const choice: string = event.detail || ''
+  if (choice === 'submit') {
+    householdId = householdId.replaceAll(' ', '')
+    if (isIdValid(householdId)) {
+      policyData.household_id = householdId
+      dispatch('closed', { choice, data: policyData })
+      hasClosed = true
+    } else {
+      setNotice('Please enter a valid Household ID')
+    }
+  } else if (choice === 'cancel') {
+    dispatch('closed', { choice })
+    hasClosed = true
+    // prevents emiiting a second 'closed' event
+  } else if (!hasClosed) {
+    dispatch('closed', { choice })
+  }
+}
+
+const isIdValid = (sanitizedId: string): number | boolean =>
```

Comment:
Why is `number` included in this return type? It seems like we want the returned value (i.e. the answer to the question "is this ID valid?") to be just a boolean.

Reply:
yeah, I was kind of confused by the ts warming since it doesn't seem like it should return -1. We can change it to just boolean but there might be an extra warning.

Reply:
the return type was determined to be `number | boolean` because that's exactly what _could_ be returned by the below expression. In the case of an empty string being passed, the expression will resolve as `"".length && "".split()...` => `0 && false` => `0` (which is a numeric value).
To resolve this, you can either change the first expression to `sanitizedId.length > 0` or you can coerce it into a boolean expression with `!!sanitizedId.length`

Reply:
Sorry, I should have included this with my previous suggested code change:
```suggestion
const isIdValid = (sanitizedId: string): boolean =>
```